glean-core/rlb/src/lib.rs

```diff
@@ -461,16 +465,15 @@ pub fn set_upload_enabled(enabled: bool) {
         let old_enabled = glean.is_upload_enabled();
         glean.set_upload_enabled(enabled);
 
-        // TODO: Cancel upload and any outstanding metrics ping scheduler
-        // task. Will happen on bug 1672951.
-
         if !old_enabled && enabled {
+            glean.start_metrics_ping_scheduler();
```

Comment:
huh, interestingly we don't do this on Kotlin or Swift.

Reply:
It's correct though, when re-enabled we should also let the MPS do its work again